### ISSUE TITLE
Why LayoutAlgoritm settings are not honored ?

### ISSUE THREAD

**Issue by lafrank** [Microsoft/automatic-graph-layout#159]
In connection to #153 I would like to ask for some help again.

When I create a new graph with the code below I try to set the above mentioned parameters from code, however it is skipped for some reason.
This is how I do this:

![image](https://user-images.githubusercontent.com/15701860/35165241-dcd6c110-fd4d-11e7-96f1-49134bbb28e0.png)

But later, when I open the property editor for the graph, I see different settings:

![image](https://user-images.githubusercontent.com/15701860/35165245-e1a92656-fd4d-11e7-9dae-5461f32fc5dd.png)

If I change the settings in that form and press OK, it works as expected and reopnening the property editor shows the values set earlier.

Can you please suggest what am I doing wrong ?

**Comment by levnach** [comment]
I don't think you are doing something wrong. To debug this I would set a breakpoint in the constructor of MdsLayoutSettings and see which code creates it.

**Comment by lafrank** [comment]
Thanks for this suggestion Lev.

I did some debugging and found, that GViewer has a readonly mdsLayoutSettings field which is initialized with default settings and upon passing a new Graph to the viewer, that Graph's LayoutAlgoritmSettings is unconditionally overwritten by that field:

![image](https://user-images.githubusercontent.com/15701860/35183049-ed09970a-fddf-11e7-81e2-c947a5ebd411.png)

What I don't understand yet, is the purpose of this _readonly_ field in GViewer and how to set it to he desired value ?

**Comment by levnach** [comment]
It seems the settings are overridden only when they are of a wrong type.

**Comment by lafrank** [comment]
Wrong type in what way ? I am not entirely sure I got your point. 
I am just passing the graphviewer a constructed Graph object for drawing with a specific LayoutAlgorithmSettings. And I have no way to set the mdsLayoutSettings readonly field of the graphviewer to match the settings of the graph I pass for drawing. Hence opening the LAyoutPropertyEditor will show the default settings.

Anyhow, I implemented the below change in GViewer and also removed the readonly qualifier from the fields :

![image](https://user-images.githubusercontent.com/15701860/35183502-f24a10f2-fde7-11e7-87d8-2fe330f2a434.png)
